awd/article.py

```diff
@@ -13,17 +18,26 @@ class Article:
     metadata and binary content for uploading to our DSpace repository.
     """
 
-    def __init__(self, doi: str) -> None:
+    def __init__(self, doi: str, metadata_url: str, content_url: str) -> None:
         """Initialize article instance.
 
         Args:
             doi: A digital object identifer (doi) for an article.
+            metadata_url: The URL for retrieving metadata records.
+            content_url: The URL for retrieving article content
         """
         self.doi: str = doi
+        self.metadata_url: str = metadata_url
+        self.content_url: str = content_url
         self.database_item = None
-        self.crossref_metadata: dict[str, Any] | None = None
-        self.dspace_metadata: dict[str, Any] | None = None
-        self.article_content: bytes | None = None
+        self.crossref_metadata: dict[str, Any]
+        self.dspace_metadata: dict[str, Any]
+        self.article_content: bytes
+
+    def process(self) -> None:
```

Comment:
If eventually a `process` method might _also_ include the uploading and reporting that is still handled in the CLI, I think this name makes sense.  But if it's going to remain as-is, mostly just handling metadata retrieval, validation, and combining, I might suggest renaming to `prepare_metadata()`.